qcodes/dataset/data_set.py

```diff
@@ -282,6 +282,8 @@ def __init__(self, path_to_db: str = None,
         self._interdeps: InterDependencies_
         self._parent_dataset_links: List[Link]
         self._data_write_queue: Queue = Queue()
+        self.data = []
```

Comment:
I noticed that the notebook shows the type as a list into a dict(after the value is cached).  Is this intentional for ordering at the beginning?

Reply:
@ianhuezo Yes that should probably just be None type